robot/impl/local_robot.go

```diff
@@ -1435,3 +1467,26 @@ func (r *localRobot) MachineStatus(ctx context.Context) (robot.MachineStatus, er
 func (r *localRobot) Version(ctx context.Context) (robot.VersionResponse, error) {
 	return robot.Version()
 }
+
+// checkMaintenanceSensorReadings ensures that errors from reading a sensor are handled properly.
+func (r *localRobot) checkMaintenanceSensorReadings(ctx context.Context,
+	maintenanceAllowedKey string, sensor resource.Sensor) (bool, error) {
+	timeout := 5 * time.Second
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	// Context timeouts on this call should be handled by grpc
+	readings, err := sensor.Readings(ctx, map[string]interface{}{})
+	if err != nil {
+		return true, errors.Errorf("error reading maintenance sensor readings. %s", err.Error())
```

Comment:
by the discussions we've had, if we timeout here we should be disabling reconfiguration. If a resource is offline we also stop reconfiguration. I think generally we can stop reconfiguration if we error here at all to be conservative